src/gui/qgsadvanceddigitizingtools.h

```diff
@@ -0,0 +1,158 @@
+/***************************************************************************
+     qgsadvanceddigitizingtools.h
+     ----------------------
+     begin                : May 2024
+     copyright            : (C) Mathieu Pellerin
+     email                : mathieu at opengis dot ch
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSADVANCEDDIGITIZINGTOOLS
+#define QGSADVANCEDDIGITIZINGTOOLS
+
+#include <QWidget>
+#include <QString>
+
+#include "qgsadvanceddigitizingdockwidget.h"
+#include "qgis_gui.h"
+#include "qgis_sip.h"
+#include "qgsmapmouseevent.h"
+
+#include <QPointer>
+
+class QgsAdvancedDigitizingDockWidget;
+class QgsDoubleSpinBox;
+class QgsMapCanvas;
+
+/**
+ * \ingroup gui
+ * \brief An abstract class for advanced digitizing tools.
+ * \since QGIS 3.40
+ */
+class GUI_EXPORT QgsAdvancedDigitizingTool : public QWidget
```

Comment:
I don't think this should be a QWidget -- I highly suspect that choice would bite us in future. Rather there should be a member which creates the widget when required instead.

```suggestion
class GUI_EXPORT QgsAdvancedDigitizingTool : public QObject
```